library/Ivoz/Provider/Domain/Model/Invoice/Invoice.php

```diff
@@ -127,14 +143,55 @@ public function isWaiting(): bool
 
     public function setNumber(?string $number = null): static
     {
-        if ($number === '') {
-            //Avoid unique key issues
-            $number = null;
-        }
+        $number = $number
```

Comment:
This will conflict with db unique key `UNIQUE KEY `invoice_number_brand` (`number`,`brandId`),`